allennlp/data/vocabulary.py

```diff
@@ -587,3 +588,33 @@ def __str__(self) -> str:
         namespaces = [f"\tNamespace: {name}, Size: {self.get_vocab_size(name)} \n"
                       for name in self._index_to_token]
         return " ".join([base_string, non_padded_namespaces] + namespaces)
+
+    def print_statistics(self, ignore_error: bool = False) -> None:
+        if self._retained_counter:
+            # Caveat: Printed info is only for part of vocabulary generated from instances. If Vocabulary
```

Comment:
Please can you print this caveat before the statistics are printed? It's moderately important.

Reply:
Sure 👍